src/query_plan/operation.rs

```diff
@@ -0,0 +1,435 @@
+                            arguments: field.arguments.clone(),
+                            directives: field.directives.clone(),
+                            selection_set: NormalizedSelectionSet {
+                                ty: field.selection_set.ty.clone(),
+                                selections: IndexMap::new(),
+                            },
+                        },
+                    )))
+                {
+                    let merged_selections = merge_selections(
+                        &field_entry.selection_set.selections,
+                        &expanded_selection_set,
+                    );
+                    let mut selection_set = field_entry.selection_set.clone();
+                    selection_set.selections = merged_selections;
+                    field_entry.make_mut().selection_set = selection_set;
+                    // field_entry.selection_set.selections = merged_selections;
+                }
+            }
+            Selection::FragmentSpread(named_fragment) => {
+                if let Some(fragment) = fragments.get(&named_fragment.fragment_name) {
+                    let expanded_selection_set = normalize_selections(
+                        &fragment.selection_set.selections.to_owned(),
+                        fragments,
+                    );
+                    normalized = merge_selections(&normalized, &expanded_selection_set);
+                } else {
+                    // no fragment found
+                }
+            }
+            Selection::InlineFragment(inline_fragment) => {
+                let expanded_selection_set = normalize_selections(
+                    &inline_fragment.selection_set.selections.to_owned(),
+                    fragments,
+                );
+
+                if let NormalizedSelection::NormalizedInlineFragment(fragment_entry) = normalized
+                    .entry(inline_fragment.into())
+                    .or_insert(NormalizedSelection::NormalizedInlineFragment(Node::new(
+                        NormalizedInlineFragment {
+                            type_condition: inline_fragment.type_condition.clone(),
+                            directives: inline_fragment.directives.clone(),
+                            selection_set: NormalizedSelectionSet {
+                                ty: inline_fragment.selection_set.ty.clone(),
+                                selections: IndexMap::new(),
+                            },
+                        },
+                    )))
+                {
+                    let merged_selections = merge_selections(
+                        &fragment_entry.selection_set.selections,
+                        &expanded_selection_set,
+                    );
+                    let mut selection_set = fragment_entry.selection_set.clone();
+                    selection_set.selections = merged_selections;
+                    fragment_entry.make_mut().selection_set = selection_set;
```

Comment:
Similar to above, I think this can be simplified to
```suggestion
                    fragment_entry.make_mut().selection_set.selections = merged_selections;
```